bin/onlyswaps-state-api/src/service.rs

```diff
@@ -0,0 +1,342 @@
+use crate::serde::ShortNumber;
+use crate::state::{AppState, SwapTransaction};
+use alloy::primitives::{Address, FixedBytes};
+use serde::Deserialize;
+use tokio::sync::watch;
+
+pub(crate) trait StateService: Send + Sync {
+    fn get_transactions(
+        &self,
+        filter: SwapTransactionQueryFilter,
+    ) -> anyhow::Result<Vec<SwapTransaction>>;
+}
+
+#[derive(Debug, Default, Deserialize)]
+pub(crate) struct SwapTransactionQueryFilter {
+    pub request_id: Option<FixedBytes<32>>,
+    pub chain_id: Option<ShortNumber>,
+    pub address: Option<Address>,
+    pub sender: Option<Address>,
+    pub recipient: Option<Address>,
+    pub solver: Option<Address>,
+}
+pub(crate) struct ChannelStateService {
+    rx: watch::Receiver<AppState>,
+}
+
+impl ChannelStateService {
+    pub fn new(rx: watch::Receiver<AppState>) -> Self {
+        Self { rx }
+    }
+}
+
+impl StateService for ChannelStateService {
+    fn get_transactions(
+        &self,
+        filter: SwapTransactionQueryFilter,
+    ) -> anyhow::Result<Vec<SwapTransaction>> {
+        let mut state = self.rx.borrow().clone();
+
+        if let Some(id) = filter.request_id {
+            by_id(&mut state.transactions, &id)
+        }
+        if let Some(chain_id) = filter.chain_id {
+            by_chain_id(&mut state.transactions, chain_id)
```

Comment:
any reason to pass by reference `&id` but latter using `chain_id` and same for all the ones below??